### DISCUSSION TITLE
Nested order by relation

### DISCUSSION THREAD

**Discussion by alexpuertasr** [prisma/prisma#7423] [Q&A] [answered]
Hi 👋 ,

I'm trying to get the chats of a specific user. Ordering this chats based on the messages. The chat with the most recent message have to be the first chat. I tried something like this but is not working:

```ts
await prisma.user
        .findUnique({
          where: { id: user.id },
        })
        .chats({
          include: { chat: { include: { messages: true } } },
          orderBy: { chat: { messages: { createdAt: 'desc' } } },
        });
```

My schema is the following:

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["orderByRelation"]
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

model User {
  id        Int        @id @default(autoincrement())
  email     String     @unique
  username  String     @unique
  firstName String?
  lastName  String?
  password  String
  chats     UserChat[]
  messages  Message[]
  createdAt DateTime   @default(now())
  updatedAt DateTime   @updatedAt
}

model UserChat {
  user   User? @relation(fields: [userId], references: [id])
  chat   Chat? @relation(fields: [chatId], references: [id])
  userId Int
  chatId Int

  @@id([userId, chatId])
}

model Chat {
  id       Int        @id @default(autoincrement())
  users    UserChat[]
  messages Message[]
}

model Message {
  id        Int      @id @default(autoincrement())
  message   String
  user      User?    @relation(fields: [userId], references: [id])
  chat      Chat?    @relation(fields: [chatId], references: [id])
  userId    Int?
  chatId    Int?
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}
```

Thanks 😊

**Comment by ryands17** [comment, marked as answer]
@alexpuertasr 👋 

Currently ordering by the many side of the relation is not yet supported. It would be great if you could add a 👍 on any of these requests that match your use case.

https://github.com/prisma/prisma/issues/6228
https://github.com/prisma/prisma/issues/5837

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
